Signal/src/AppDelegate.m

```diff
@@ -819,6 +817,16 @@ - (void)databaseViewRegistrationComplete
 
     if ([TSAccountManager isRegistered]) {
         DDLogInfo(@"localNumber: %@", [TSAccountManager localNumber]);
+
+        // Fetch messages as soon as possible after launching. In particular, when
+        // launching from the background, without this, we end up waiting some extra
+        // seconds before receiving an actionable push notification.
+        [[Environment getCurrent].messageFetcherJob runAsync];
+
+        // This should happen at any launch, background or foreground.
+        __unused AnyPromise *promise = [OWSSyncPushTokensJob runWithPushManager:[PushManager sharedManager]
```

Comment:
I thought it was unsafe to do this before `didRegisterUserNotificationSettings:`.  Maybe we should do this in both places, i.e.:

* Start `OWSSyncPushTokensJob` in `didRegisterUserNotificationSettings:` if database view registration is complete.
* Start `OWSSyncPushTokensJob` in `databaseViewRegistrationComplete:` if `didRegisterUserNotificationSettings` has already been called (we'll need a new flag).

Reply:
SyncPushTokensJob already makes sure to call `registerUserNotificationSettings`

Reply:
This is more firmly established in the followup PR #2620

Reply:
Or maybe we can have a `tryToSyncPushTokens` method that we can call from both places that only works if both have completed.